### PR TITLE
Support backgrounding audio and video

### PR DESCRIPTION
**Pull request by armstrongnate**
...from FileDetailsViewController.

refs: MBL-14228
affects: student
release note: Added support for continuing audio and video while the app is in the background

Test plan:
* See ticket for an assignment with links to audio and video files
* The audio and video should still play when the app gets backgrounded
* The audio and video should stop when you tap the back button

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/eayf4n3) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Feayf4n3)
teacher: [Link](https://rebrand.ly/8bm69w4) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F8bm69w4)
parent: [Link](https://rebrand.ly/r2yv8jb) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fr2yv8jb)

**Comment by inst-danger** [comment]
#### Release Note: 
Added support for continuing audio and video while the app is in the background

#### Affected Apps: Student

[MBL-14228](https://instructure.atlassian.net/browse/MBL-14228)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.24% | 93.26% | -0.03%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/AppEnvironment/AppEnvironment.swift [hunk 1]
```diff
@@ -68,4 +72,23 @@ open class AppEnvironment {
...
+
+    public func reconnectVideoPlayer() {
+        videoPlayerViewController?.player = videoPlayer
+        videoPlayerViewController = nil
+    }
+
+    public func disconnectVideoPlayer() {
+        videoPlayerViewController = nil
+        videoPlayer = nil
+    }
```
Comment (bluwave) [inline Core/Core/AppEnvironment/AppEnvironment.swift:93]: Is this the best place for this?  It seems easy and convenient to put here but AppEnvironment seems to be picking up all things at a good pace.  Would it be better to have some AudioVideo State singleton.  Just throwing that out there.  I'm not necessarily opposed to to it here either.

**Comment by armstrongnate** [comment]
I was surprised, too, @vanwagonet. For reference, I'm following these docs: https://developer.apple.com/documentation/avfoundation/media_assets_playback_and_editing/creating_a_basic_video_player_ios_and_tvos/playing_audio_from_a_video_asset_in_the_background

To get audio to work all needed was the background audio entitlement.

**Comment by armstrongnate** [comment]
I'll move this stuff from AppEnvironment to a new `AudioVideo` helper.

File: Core/Core/AppEnvironment/AppEnvironment.swift [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 import Foundation
 import CoreData
+import AVKit
```
Comment (bluwave) [inline Core/Core/AppEnvironment/AppEnvironment.swift:21]: Probably don't need this here any longer.

**Pull request merged**